test/strings/basic.jl

```diff
@@ -461,3 +460,7 @@ Base.endof(x::CharStr) = endof(x.chars)
 # Case with Unicode characters
 @test cmp("\U1f596\U1f596", CharStr("\U1f596")) == 1   # Gives BoundsError with bug
 @test cmp(CharStr("\U1f596"), "\U1f596\U1f596") == -1
+
+# issue #12495: logical indexing attempt raises MethodError
```

Comment:
Good catch, I first thought about throwing `MethodError` but then realized it would be confusing. Fixed.